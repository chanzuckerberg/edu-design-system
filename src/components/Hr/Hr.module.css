/*------------------------------------*\
    # HR (HORIZONTAL RULE)
\*------------------------------------*/

/**
 * Line that acts as a separator of content.
 * 
 */
.hr {
  margin-top: calc(var(--eds-spacing-size-4) / 16 * 1rem);
  margin-bottom: calc(var(--eds-spacing-size-4) / 16 * 1rem);
  border-top: none;
  border-right: none;
  border-left: none;
  border-bottom-width: calc(var(--eds-border-width-sm) * 1px);
  border-bottom-style: solid;
  border-bottom-color: var(--eds-theme-color-border-neutral-subtle);
}

/**
 * Horizontal rule with a thick border.
 */
.hr--lg {
  border-width: calc(var(--eds-border-width-lg) * 1px);
}

/**
 * Branded horizontal rule.
 */
.hr--brand {
  border-color: var(--eds-theme-color-border-brand-primary-strong);
}