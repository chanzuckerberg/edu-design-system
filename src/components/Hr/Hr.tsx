import clsx from 'clsx';
import React from 'react';
import styles from './Hr.module.css';

export interface Props {
  /**
   * Stylistic variations:
   * - **brand** yields a branded horizontal rule
   */
  variant?: 'brand';

  /**
   * Size variations:
   * - **lg** yields a thicker horizontal rule
   */
  size?: 'lg';
  /**
   * CSS class names that can be appended to the component.
   */
  className?: string;
}

/**
 * Primary UI component for user interaction
 */
export const Hr = ({ className, size, variant, ...other }: Props) => {
  const componentClassName = clsx(
    styles['hr'],
    className,
    size === 'lg' && styles['hr--lg'],
    variant === 'brand' && styles['hr--brand'],
  );

  return <hr className={componentClassName} {...other} />;
};
