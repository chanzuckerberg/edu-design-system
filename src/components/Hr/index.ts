export { Hr as default } from './Hr';
