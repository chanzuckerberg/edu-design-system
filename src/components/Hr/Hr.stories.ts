import type {StoryObj, Meta} from '@storybook/react';
import type {ComponentProps} from 'react';

import {Hr} from './Hr';

export default {
  title: 'Atoms/Text/Hr',
  component: Hr,
} as Meta<Args>;

type Args = ComponentProps<typeof Hr>;

export const Default: StoryObj<Args> = {};

export const Large: StoryObj<Args> = {
  args: {
    size: 'lg',
  },
};

export const Brand: StoryObj<Args> = {
  args: {
    variant: 'brand',
  },
};
