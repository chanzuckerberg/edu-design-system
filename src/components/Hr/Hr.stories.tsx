import React from 'react';
import { Story, Meta } from '@storybook/react';

import { Hr, Props } from './Hr';

export default {
  title: 'Atoms/Text/Hr',
  component: Hr,
} as Meta;

const Template: Story<Props> = (args) => <Hr {...args} />;

export const Default = Template.bind({});
Default.args = {};

export const Large = Template.bind({});
Large.args = {
  size: 'lg',
};

export const Brand = Template.bind({});
Brand.args = {
  variant: 'brand',
};
