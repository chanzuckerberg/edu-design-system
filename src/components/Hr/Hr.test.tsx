import { generateSnapshots } from '@chanzuckerberg/story-utils';
import { render } from '@testing-library/react';
import React from 'react';

import { Hr } from './Hr';

import * as stories from './Hr.stories';
import type { StoryFile } from '../../../.storybook/utility-types';

describe('<Hr />', () => {
  generateSnapshots(stories as StoryFile);

  describe('emits messages when misused', () => {
    let consoleWarnMock: jest.SpyInstance;
    beforeEach(() => {
      // Add in mocks for the calls that can occur in implementation to suppress logging in tests
      consoleWarnMock = jest.spyOn(console, 'warn');
      consoleWarnMock.mockImplementation();
    });

    afterEach(() => {
      jest.resetAllMocks();
    });

    it('warns when Hr instances use size prop', () => {
      render(<Hr size="lg" />);

      expect(consoleWarnMock).toHaveBeenCalledTimes(1);
    });

    it('warns when Hr instances use variant prop', () => {
      render(<Hr variant="brand" />);

      expect(consoleWarnMock).toHaveBeenCalledTimes(1);
    });
  });
});
