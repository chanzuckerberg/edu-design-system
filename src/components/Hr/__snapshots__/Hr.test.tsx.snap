// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hr /> Default story renders snapshot 1`] = `
<hr
  class="hr w-[384px]"
/>
`;
